Draft of blog post on how we use the rOpenSci in the Master of Data Science program

ttimbers opened the pull request.

ttimbers commented:
cc @stefaniebutland

noamross commented on content/blog/2019-08-05-software-peer-review-guidelines-for-teaching.md:
```suggestion
This is a project-based course where students are required to work together in teams of three or four. Here the project involves developing a Python and an R software package. The two packages are expected to do essentially the same thing. They are asked to make one in each language so that they can learn how to do both, and learn to generalize about software packages. For the project, the students are guided to choose the package topic from one of the following themes: 1) functions that are entirely new to the R or Python ecosystem, 2) improve upon pre-existing functions in either language or 3) re-implement existing code/functions that they wish to deepen their understanding of (*e.g.,* write a linear regression package from scratch). Students are asked to place emphasis in their work on robust software engineering (*e.g.,* continuous integration, testing, documentation, licensing) and collaboration (*e.g.,* advanced version control).
```

Perhaps too many "e.g."'s?  Consider a little more variety in structure.

noamross commented on content/blog/2019-08-05-software-peer-review-guidelines-for-teaching.md:
```suggestion
For courses in the Master of Data Science program we work hard to find and curate high quality and current resources for our students to help them in their coursework. For this collaborative software development course, one of the best resources we recommend for the R packages is the [rOpenSci Packages: Development, Maintenance, and Peer Review](https://devguide.ropensci.org/) guide. In particular, we encourage the students to go through the [Review template](https://devguide.ropensci.org/reviewtemplate.html) in the appendix of that guide before they hand in their R package for final grading. We recommend this, as by its design and use in the wild, it is a comprehensive checklist to review if an R package is of high quality and complete (in regards to software development). Instructors and Teaching Assistants for this course also use this [Review template](https://devguide.ropensci.org/reviewtemplate.html) to help give both formative feedback and guide summative assessment. We also use the rOpenSci guide to to develop formative feedback and summative assessment for the Python packages, though the [in-development guide from pyOpenSci](https://www.pyopensci.org/dev_guide) may play this role. 
```

The pyOpenSci community is developing a guide! So I revised the last couple of sentences here.

rcbrett replied:
perfect thanks for revising those changes!

ttimbers replied:
perfect, thanks for suggesting those revisions!

noamross commented on content/blog/2019-08-05-software-peer-review-guidelines-for-teaching.md:
```suggestion
We think that by using the [rOpenSci Packages: Development, Maintenance, and Peer Review](https://devguide.ropensci.org/) guide in our [DSCI 524 Collaborative Software Development](https://github.com/UBC-MDS/DSCI_524_collab-sw-dev) course in the [University of British Columbia's Master of Data Science program](https://ubc-mds.github.io/about/) we can better teach our students how to create high quality software packages, and expose them to the idea of package review in hopes that they will embrace this culture when they create their own Data Science software packages out in the wild in their future careers. We are grateful to rOpenSci for making this guide open. Although this guide was not intended/built for an educational use case, we have found it is very useful for our Collaborative Software Development course. We hope this story of our use case of this guide can show (or remind) others that open resources can become more valuable for the broader community as they find new uses.
```

noamross commented:
Great work! My remaining comment is that I thought you'd want to mention the value of _your_ open resources, as well! You link to the course lot but don't mention that what is being linked to is the an open syllabus and resources, at that these are available for all your courses. Your call.

stefaniebutland commented:
@ttimbers I second Noam in encouraging you to take the opportunity to highlight the value of _your_ resources!

ttimbers commented:
@stefaniebutland - changes incorporated! Thanks to you and @noamross for the opportunity to write and post this on the ropensci blog and for your help and feedback with the post. It was a pleasure!

karthik commented:
@ttimbers You created a second roweb2 directory in your copy. That's why the author info is messed up. You need to move that to `content/author/...` and delete your roweb2 folder that inside roweb2. Does that make sense?

stefaniebutland commented:
@ttimbers I asked Karthik to help me since I couldn't see a version showing your last commits. In addition to his author file request above, please set
- in YAML `date: 2019-08-27`
- filename `2019-08-05-software-peer-review-guidelines-for-teaching.md`
- confirm 2 typos fixed
- confirm accept/reject suggestions of headings and moving/omitting course number

It's strange that I was able to see things before, but with your date change to 2019-08-05 in last commits I lost ability to see/find current version. Thanks for looking into this

ttimbers commented:
@stefaniebutland - apologies for the filename and folder duplication mistakes. I have fixed those now, and think I have addressed all other suggested changes.

stefaniebutland commented:
Thank you @ttimbers! It's good to go on the 27th
